chainer/functions/util/forget.py

```diff
@@ -43,12 +42,16 @@ def forward(self, inputs):
 
     def backward(self, indexes, grad_outputs):
         inputs = self.get_retained_inputs()
+        # Create new variables that have no creators
+        dummy_inputs = tuple([variable.Variable(inp.array) for inp in inputs])
+
         with function.force_backprop_mode():
-            outs = _call_func(self.func, inputs)
-        # Return gradients that are further backproable
-        return chainer.grad(
-            outs, inputs, grad_outputs=grad_outputs,
-            enable_double_backprop=True)
+            outs = _call_func(self.func, dummy_inputs)
+        assert len(outs) == len(grad_outputs)
+        for out, grad_output in zip(outs, grad_outputs):
+            out.grad_var = grad_output
+            out.backward()
```

Comment:
When there are N>1 outputs, it looks backward is called N times, which is redundant when these outputs share a part of the computational graph. Could you reduce it to only one backward call? For example, all the paths are covered using `F.identity`:

```py3
outs = F.identity(*outs)
for out, grad_output in zip(outs, grad_outputs):
    out.grad_var = grad_output
outs[0].backward()  # other outputs are taken into account through the backward of F.identity
```

Reply:
Thank you for the suggestion. Unfortunately, the code you suggested didn't work well. I couldn't fix it. One error is that the test `TestForgetGrad` didn't pass.

Reply:
Oh, I noticed that we need to call `F.identity` in `with function.force_backprop_mode():` block. In that case, your code works well.

Reply:
Ahhh, sorry for my incomplete code.